Sample/Kitchen Sink/DemonstrationTools/UnitConverter.swift

```diff
@@ -0,0 +1,184 @@
+//
+//  Shape+Extensions.swift
+//  flight-ui-ios
+//
+//  Created by Appivate 2023
+//
+
+import Foundation
+import SwiftUI
+import FlightUI
+
+struct UnitConverter: View {
+    @EnvironmentObject var theme: Theme
+    @StateObject var themeManager = ThemeManager(current: .dark)
+    @ObservedObject var demonstrationVM = UnitConverterViewModel()
+    @State private var weightValuesSwapped = false
+
+    var body: some View {
+        ScrollView {
+
+            VStack {
+                HeadingView(
+                    title: demonstrationVM.weightTitle,
+                    subTitle: demonstrationVM.weightSubtitle)
+
+                HStack {
+                    if  weightValuesSwapped {
+                        kgInputField
+                        swapButton
+                        lbInputField
+                    } else {
+                        lbInputField
+                        swapButton
+                        kgInputField
+                    }
+                    convertButton1
+                }
+            }
+            .padding([.top], themeManager.current.padding.grid2x)
+
+            HStack {
+                VStack {
+                    HeadingView(
+                        title: demonstrationVM.adjustableConversionTitle,
+                        subTitle: demonstrationVM.adjustableConversionSubTitle)
+                    HStack {
+                        lengthInput
+                        lengthInputUnitPicker
+                    }
+
+                    HStack {
+                        calculationResult
+                        lengthOutputUnitPicker
+                    }
+                }
+                VStack {
+                    Spacer()
+                    convertButton2
+                        .padding([.bottom], themeManager.current.padding.grid2x)
+                }
+            }
+            .padding([.top], themeManager.current.padding.grid2x)
+        }
+        .padding([.leading, .trailing], themeManager.current.padding.grid2x)
+        .background(theme.color.background)
+        .navigationBarTitle(demonstrationVM.naivgationBarTitle)
+    }
+
+    var calculationResult: some View {
+        InputField(text: $demonstrationVM.outputValue, placeholder: demonstrationVM.calculatedField)
+            .textFieldStyle(.advisory)
+            .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var lengthInput: some View {
+        InputField(text: $demonstrationVM.inputValue,
+                   placeholder: demonstrationVM.lengthPalceholder,
+                   bottomLabelConfig: BottomLabelConfig(""),
+                   filter: .doubleOnly)
+        .textFieldStyle(.default)
+    }
+
+    var lengthInputUnitPicker: some View {
+        MenuField(selection: $demonstrationVM.boundSelectionInput,
+                  options: UnitConverterViewModel.LengthType.allCases,
+                  placeholder: "")
```

Comment:
Super minor: `placeholder` has a default value which will show no placeholder text so doesn't need to be provided, the length menus also selections that have a default value so will never be used though 👍